src/main/java/utils/FileStream.java

```diff
@@ -22,67 +23,69 @@ public static Path getFileName(Path filePath, String ending) {
     }
 
     //create file by his name
-    public static FileWriter createFile(Path fileName) throws IOException {
-        //create the decrypted file
-        File file;
-        file = new File(fileName.toString());
-        if (!file.createNewFile()) {
-                PrintWriter writer = new PrintWriter(file);
-                writer.print("");
-                writer.close();
-            }
-
-        //create the file-writer for the file
-        return new FileWriter(file.getAbsoluteFile());
+    public static File createFile(Path fileName) throws IOException {
+       try {
+           //create the decrypted file
+           File file;
+           file = new File(fileName.toString());
+           if (!file.createNewFile()) {
+               PrintWriter writer = new PrintWriter(file);
+               writer.print("");
+               writer.close();
+           }
+
+           //create the file-writer for the file
+           return file;
+       }
+       catch (IOException ex) {
+           throw new IOException("The function can't succeed to create file. \n File: "+fileName.getFileName() +"\n Error: "+ex.getMessage());
+       }
     }
 
     //get the key file
-    public static List<Integer> getListOfIntegers(Path keyFilePath) throws FileNotFoundException {
-
-        //read the key from the file
+    public static List<Integer> getListOfIntegers(Path keyFilePath) throws FileNotFoundException, InvalidEncryptionKeyException {
```

Comment:
```suggestion
    public static List<Integer> readKeys(Path keyFilePath) throws FileNotFoundException, InvalidEncryptionKeyException {
```
המטרה היא שהפונקצייה תסביר את עצמה. אם המטרה היא לקרוא מפתחות -  החתימה שלה כבר מבהירה שמדובר בint

Reply:
done